XUtils.h

```diff
@@ -76,4 +76,7 @@ char* xStrndup(const char* str, size_t len) ATTR_NONNULL ATTR_MALLOC;
 ssize_t xReadfile(const char* pathname, void* buffer, size_t count);
 ssize_t xReadfileat(openat_arg_t dirfd, const char* pathname, void* buffer, size_t count);
 
+/* This subtraction is used by NetBSD / OpenBSD for calculation of CPU usage items. */
+unsigned long long saturatingSub(unsigned long long a, unsigned long long b);
```

Comment:
Since this is such a trivial function, it's probably worth defining it `inline`.
Another option might be to implement it as macro in `Macros.h`, to be type generic.
Thoughts @BenBE ?

p.s.: it's used also at https://github.com/htop-dev/htop/blob/323d7e73aa1cc13b3a0870530e8fbd27102259ec/linux/LinuxProcessList.c#L1809

Reply:
I think we should prefer a type-safe implementation over the macro one, especially as this automagically does CSE and avoids side-effects due to re-evaluation of arguments as the macro variant would do. The only ever implementations we probably need to have are for `int32_t`, `int64_t` and `double`. And while `XUtils.[ch]` primarily focuses on memory/string stuff, I wonder if the fit into `Macros.h` would be much better. Nah, well, make it `static inline` and put it in `Macros.h` ;-)

Reply:
Do you want me to shift the function definition to `Macros.h` and make it `static inline`?

Reply:
Should be done in c68bab1

Reply:
Please also consolidate with the location/macro that @cgzones mentioned, if not yet done. Else feel free to mark this one here as resolved.

Reply:
You mean the one for `LinuxProcessList.c` too?

Reply:
Yes.

Reply:
This should be done in 1e279c6